docs/root/start/sandboxes/single-page-app.rst

```diff
@@ -0,0 +1,597 @@
+.. _install_sandboxes_single_page_app:
+
+Single page React app (with OAuth)
+==================================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+        Used to make HTTP requests.
+
+   :ref:`envsubst <start_sandboxes_setup_envsubst>`
+        Used to interpolate environment vars in templates.
+
+   :ref:`jq <start_sandboxes_setup_jq>`
+        Used to parse JSON.
+
+   :ref:`mkpasswd <start_sandboxes_setup_mkpasswd>`
+        Used to generate a ~random HMAC token.
+
+This sandbox provides an example of building and developing a single page app with Envoy.
+
+The sandbox covers a number of Envoy's features, including:
+
+- :ref:`direct_response <config_network_filters_direct_response>`
+- :ref:`OAuth <config_http_filters_oauth>`
+- Dynamic xDS filesystem updates
+- Websocket proxy
+- Gzip :ref:`compression <config_http_filters_compressor>`
+- TLS/SNI up/downstream connection/termination
+- Path/host rewrites
+
+The app is built with `React <react_>`__ using `Vite <vite_>`__ and demonstrates OAuth authentication using
+Envoy's :ref:`OAuth filter <config_http_filters_oauth>`.
+
+This covers a scenario where we want OAuth to both authenticate the user and provide credentials
+for further API interactions.
+
+This is enabled by setting the OAuth configuration
+:ref:`forward_bearer_token <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.forward_bearer_token>`
+to ``true``
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 36-40
+   :linenos:
+   :lineno-start: 36
+   :emphasize-lines: 3
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+.. warning::
+   Setting
+   :ref:`forward_bearer_token <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.forward_bearer_token>`
+   means the provided access token will be forwarded to upstreams proxied by Envoy unless explicitly excluded.
```

Comment:
This is a good warning, but what are the consequences? Presumably the point is you must trust upstream proxies, and if they issued the bearer token in the first place it's OK?

Reply:
if you have other proxies set up in envoy and you have the cookie - it will be sent to any upstreams that are not excluded

Reply:
```
... will be forwarded to all other upstreams for the same listener and filter chain, including those that did not issue the original token. 

This behavior can be avoided by configuring the :ref:`pass_through_matcher <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.pass_through_matcher>` or by disabling the OAuth2 filter with per-route configuration (link to https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpfilter-disabled ).
```

Reply:
I think the per-route configuration is more natural for Envoy, I might just drop the reference to pass_through_matcher TBH.

Reply:
Although I wonder if this should be there to strip the headers from the client in. the case you don't want to pass through? In any case, having better per-route control over this filter would have made sense.

Reply:
Actually, reading the code, pass_through_matcher isn't what you want. It will forward whatever the client sends, which includes bearer token. You'd want to remove the header somehow with RouteConfiguration mutations to avoid forwarding Authorization or bearer tokens to untrusted upstreams.

Reply:
`pass_through_matcher` needs to be set anyway - not sure re token - if i dont include `/hub*` path in the pass through it doesnt work

originally i had thought to add something about scrubbing headers, but then i noticed that it wasnt passing through - probs should check this again

Reply:
to clarify - if i dont add `hub*` to **inverted** `pass_through_matcher` - ie so it _is_ passed through it doesnt get the creds afaict

or put another way anything that is passed through gets no cookies/creds

Reply:
testing this further, it seems `pass_through_matcher` has no effect on creds they are passed through regardless

config doesnt need `hub.*` in the matcher

i will update/remove the ref to p_t_m here

i would rather not go too far into how to scrub the cookies/auth but would be good to ref something